dsl_parser/functions.py

```diff
@@ -633,6 +633,79 @@ def evaluate(self, handler):
         return results
 
 
+@register(name='get_attributes_dict', func_eval_type=RUNTIME_FUNC)
+class GetAttributesDict(Function):
+
+    def __init__(self, args, **kwargs):
+        self.node_name = None
+        self.attribute_path = None
+        super(GetAttributesDict, self).__init__(args, **kwargs)
+
+    def parse_args(self, args):
+        self.node_name = args[0]
+        self.attribute_paths = args[1:]
+        for pos, attribute_path in enumerate(self.attribute_paths):
+            # These must all be lists due to the way the attribute_path
+            # processing functions work
+            if not isinstance(attribute_path, list):
+                self.attribute_paths[pos] = [attribute_path]
+
+        for attribute_path in self.attribute_paths:
+            check = [self.node_name] + attribute_path
+            if not _is_legal_nested_attribute_path(check):
+                raise ValueError(
+                    'Illegal arguments passed to {0} function. Expected: '
+                    '[ node_name\\function, list of attribute_name\\function '
+                    'or [attribute_name\\function, '
+                    'nested-attribute-1\\function], ... ] '
+                    'but got: {1}.'.format(self.name, args))
+
+        duplicate_or_ambiguous_check = [
+            self._convert_to_identifier(attribute_path)
+            for attribute_path in self.attribute_paths
+        ]
+        duplicated_or_ambiguous = [
+            item for item in duplicate_or_ambiguous_check
+            if duplicate_or_ambiguous_check.count(item) > 1
+        ]
+        if duplicated_or_ambiguous:
+            raise exceptions.FunctionEvaluationError(
+                'One or more duplicated or ambiguous attribute paths found: '
+                '{}'.format(', '.join(duplicated_or_ambiguous))
```

Comment:
2.6 for ci?

Reply:
Actually just passed, strangely enough. Given that we want to remove 2.6 for this anyway, is it worth changing?

Reply:
huh. Then no.